controller/controller_server/csi_controller_server.py

```diff
@@ -60,14 +61,21 @@ def CreateVolume(self, request, context):
         source_type, source_id = self._get_source_type_and_id(request)
 
         logger.debug("Source {0} id : {1}".format(source_type, source_id))
-        try:
-            secrets = request.secrets
-            user, password, array_addresses = utils.get_array_connection_info_from_secret(secrets)
 
-            pool = request.parameters[config.PARAMETERS_POOL]
+        topologies = utils.get_volume_topologies(request)
 
-            space_efficiency = request.parameters.get(config.PARAMETERS_SPACE_EFFICIENCY)
+        secrets = request.secrets
 
+        try:
+
+            user, password, array_addresses, secret_uid = utils.get_array_connection_info_from_secrets(
+                secrets=secrets,
+                topologies=topologies)
+            logger.info("chosen array_addresses: {}".format(array_addresses))
```

Comment:
Usually, we don't write the parameter name (`array_addresses`) in the logs, but a description of the formatted parameter.
In this case:
`array addresses`

Reply:
removed